### PR TITLE
[DEV-2318] observation_table: validate entities are present on creation

### PR DESCRIPTION
**Pull request by jevonyeoh**
## Description
Add validation into the observation_table creation path to ensure that entities are present when creating the observation table. 

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: tests/integration/api/test_observation_table.py [hunk 1]
```diff
@@ -96,15 +112,15 @@ async def test_observation_table_from_view(event_table, scd_table, session, sour
...
-        columns_rename_mapping={view.timestamp_column: "POINT_IN_TIME"},
+        columns_rename_mapping={view.timestamp_column: "POINT_IN_TIME", "ÜSER ID": "üser id"},
     )
     assert observation_table.name == f"MY_OBSERVATION_TABLE_FROM_VIEW_{source_type}"
     check_location_valid(observation_table, session)
     await check_materialized_table_accessible(observation_table, session, source_type, sample_rows)
 
     check_materialized_table_preview_methods(
         observation_table,
-        expected_columns=["POINT_IN_TIME", "ÜSER ID"],
```
Comment (jevonyeoh) [inline tests/integration/api/test_observation_table.py:107]: @YS-L - with this validation, we will also be enforcing case sensitivity in column names. how do you feel about it?

File: featurebyte/service/observation_table.py [hunk 2]
```diff
@@ -184,6 +186,10 @@ async def validate_materialized_table_and_get_metadata(
                 f"Point in time column should have timestamp type; got {point_in_time_dtype}"
             )
 
+        # Check that there's at least one entity mapped in the columns info
+        if not any(info.entity_id is not None for info in columns_info):
+            raise ValueError("At least one entity column should be provided.")
```
Comment (YS-L) [inline featurebyte/service/observation_table.py:191]: Observation table without any entity is valid for for features created without entity, though it should be rare that a feature list only contains such features. Curious what is the rationale of enforcing this new rule?

Comment (YS-L) [reply featurebyte/service/observation_table.py:191]: I think the validation makes sense as it avoids potential user error in most cases other than the edge case I mentioned. Is this validation applicable for `BatchRequestTable` as well?

File: tests/integration/api/test_observation_table.py (hunk 1 above)
Comment (YS-L) [reply tests/integration/api/test_observation_table.py:107]: That seems fine to me since it should be a requirement already anyway.

**Review by YS-L — APPROVED** [review]

File: featurebyte/service/observation_table.py (hunk 2 above)
Comment (jevonyeoh) [reply featurebyte/service/observation_table.py]: as discussed in person, reason for implementing this is to ensure the observation tables have some entity column that can be used for serving. 

this is only for OT's created from source tables, views and targets.
